granulate_utils/linux/cgroups/base_controller.py

```diff
@@ -0,0 +1,54 @@
+#
+# Copyright (c) Granulate. All rights reserved.
+# Licensed under the AGPL3 License. See LICENSE.md in the project root for license information.
+#
+
+from __future__ import annotations
+
+from pathlib import Path
+from typing import Optional, Set, Union
+
+from granulate_utils.linux.cgroups.cgroup import (
+    CgroupCore,
+    create_cgroup_from_path,
+    get_current_process_cgroup,
+    is_known_controller,
+)
+
+
+class BaseController:
+    controller: str
+
+    def __init__(self, cgroup: Optional[Union[Path, CgroupCore]] = None) -> None:
+        assert is_known_controller(self.controller), f"{self.controller!r} is not supported"
```

Comment:
`self.controller` was no initialized at this point?

Reply:
Oh, this is a class attribute?

Reply:
Maybe comment for clarity

Reply:
yes it's a class attribute, i'll add a comment